src/shared/api/services/workspace.service.ts

```diff
@@ -0,0 +1,32 @@
+import { axiosService } from '~/shared/api';
+import {
+  GetWorkspaceAppletRespondentPayload,
+  GetWorkspaceAppletRespondentSuccessResponse,
+  GetWorkspaceRoles,
+  GetWorkspaceRolesSuccessResponse,
+} from '~/shared/api/types/workspace';
+
+function workspaceService() {
+  return {
+    getWorkspaceRoles(payload: GetWorkspaceRoles) {
+      return axiosService.get<GetWorkspaceRolesSuccessResponse>(
+        `/workspaces/${payload.workspaceId}/roles`,
+        {
+          params: payload.appletIds
+            ? {
+                appletIds: payload.appletIds,
+              }
+            : undefined,
+        },
+      );
+    },
+
+    getWorkspaceAppletRespondent(payload: GetWorkspaceAppletRespondentPayload) {
+      return axiosService.get<GetWorkspaceAppletRespondentSuccessResponse>(
+        `/workspaces/${payload.workspaceId}/applets/${payload.appletId}/respondents/${payload.respondentId}`,
+      );
+    },
```

Comment:
> I'm not sure this is it. The workspace ID is present in the applet data. The malformed requests you're seeing are a result of the nature of hooks. The hooks have to run even before the proper data is present, so some of them will be malformed at first. Let me do some investigation to see what this could be

Looks like the issue of the malformed requests is due to missing `payload.workspaceId` at this point, but I also get your comment about hooks having to render even if missing data. 

Can we return something else here to avoid those malformed requests? e.g. `Promise.reject()` or similar

Reply:
I forgot that we can turn these off. I just turned off the queries until the `workspaceId` is available

Reply:
Much better this way, nice work